zyte_common_items/processors.py

```diff
@@ -104,50 +107,80 @@ def _from_zp_breadcrumb(value: zp_Breadcrumb) -> Breadcrumb:
+def brand_processor(value: Any, page: Any) -> Any:
     """Convert the data into a brand name if possible.
 
-    Supported inputs are :class:`~parsel.selector.Selector`,
-    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
-    Other inputs are returned as is.
+    If inputs are either :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` or :class:`~lxml.html.HtmlElement`, attempts
+    to extract brand data from it.
+
+    If value is a string, uses it to create a :class:`~zyte_common_items.Brand` instance.
+
+    Other inputs are returned unchanged.
     """
-    return extract_brand_name(value, search_depth=2)
+    value = _handle_selectorlist(value)
 
+    if isinstance(value, str):
+        return Brand(name=value) if value else None
 
-@only_handle_nodes
-def price_processor(value: Union[Selector, HtmlElement], page: Any) -> Any:
+    if isinstance(value, (Selector, SelectorList, HtmlElement)):
+        if brand_name := extract_brand_name(value, search_depth=2):
+            return Brand(name=brand_name)
+        else:
+            return None
+
+    return value
+
+
+def price_processor(value: Any, page: Any) -> Any:
     """Convert the data into a price string if possible.
 
     Uses the price-parser_ library.
 
     Supported inputs are :class:`~parsel.selector.Selector`,
-    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    :class:`~parsel.selector.SelectorList`, :class:`~lxml.html.HtmlElement`, string
+    instances and numeric values.
+
     Other inputs are returned as is.
 
     Puts the parsed Price object into ``page._parsed_price``.
 
     .. _price-parser: https://github.com/scrapinghub/price-parser
     """
-    price = extract_price(value)
-    page._parsed_price = price
-    return _format_price(price)
+    value = _handle_selectorlist(value)
+
+    if isinstance(value, Real):
+        return f"{value:.2f}"
+    elif isinstance(value, (Selector, HtmlElement, str)):
+        price = extract_price(value)
+        page._parsed_price = price
+        return _format_price(price)
+    else:
+        return value
 
 
-@only_handle_nodes
-def simple_price_processor(value: Union[Selector, HtmlElement], page: Any) -> Any:
+def simple_price_processor(value: Any, page: Any) -> Any:
     """Convert the data into a price string if possible.
 
     Uses the price-parser_ library.
 
     Supported inputs are :class:`~parsel.selector.Selector`,
-    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
+    :class:`~parsel.selector.SelectorList`, :class:`~lxml.html.HtmlElement`, string
+    instances and numeric values.
+
     Other inputs are returned as is.
 
     .. _price-parser: https://github.com/scrapinghub/price-parser
     """
-    price = extract_price(value)
-    return _format_price(price)
+    value = _handle_selectorlist(value)
+
+    if isinstance(value, Real):
+        return f"{value:.2f}"
```

Comment:
What do you think about moving it to https://github.com/zytedata/zyte-parsers/blob/main/zyte_parsers/price.py?

Reply:
It seems this would allow to remove the duplication in price_processor vs simple_price_processor.

Reply:
We'd like to migrate our stuff quickly before certain deadlines, which is why I want to have this thing working for now and avoid coordinating between PR in yet another dependency repository (which is also why I placed proposed parsing of images into TODO). I think we can improve it later and migrate into more fitting place.

Reply:
Sounds good!